aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -60,8 +60,9 @@
 import software.amazon.awssdk.services.dynamodb.model.TransactionConflictException;
 
 /**
- * DynamoDB implementation for the lock manager.
+ * @deprecated will be removed in 0.15.0, please use {@link org.apache.iceberg.aws.dynamodb.DynamoDbLockManager}
  */
+@Deprecated
 class DynamoLockManager extends LockManagers.BaseLockManager {
```

Comment:
Why deprecate this if it wasn't public in the first place?

Also, can this extend `DynamoDbLockManager` so that we don't have two implementations? Then we can also get rid of the tests here.

Reply:
Will let @jackye1995  confirm but I believe org.apache.iceberg.aws.glue.DynamoLockManager is public, users can pass it in as a locking implementation when creating a catalog.

+1 to extending the DynamoDBLockManager

Reply:
If this is already effectively public because people use it dynamically as their lock-impl, then I'd probably keep this instead of renaming. Or maybe have both names, but definitely not remove this unless it's absolutely necessary.

Reply:
my intention for doing that copy was to fully remove this class after 2 releases, and people who see this class can know it should not be used anymore. But extending also works, I don't have a strong opinion of which way is better.

Reply:
cool, I will update with the extending approach

Reply:
Are you sure there's much value in renaming Dynamo -> DynamoDb?

Reply:
The class path is already changing, so I thought it's better to make things consistent, as I said in the description

> Note that the new class is DynamoDbLockManager instead of DynamoLockManager to follow DynamoDb prefix convention (because the new Dynamo client in AWS SDK v2 is named DynamoDbClient).

but no strong opinion here. Any thoughts?

Reply:
Since this is something that people refer to directly already, I don't see much value in changing it. We can make the underlying classpath change transparent to users, so I'd probably opt to do that.

Reply:
> We can make the underlying classpath change transparent to users

what do you mean by transparent? Do you mean to just make `org.apache.iceberg.aws.glue.DynamoLockManager` public instead of having `org.apache.iceberg.aws.dynamodb.DynamoDbLockManager`?

Reply:
Yeah, it's okay if the library we rely on changes. If we don't make the corresponding change then users don't have to modify configurations. That's why I'd prefer to just leave this as-is.